### DISCUSSION TITLE
What dataset was medspacy trained on?

### DISCUSSION THREAD

**Discussion by Samuel-Fipps** [medspacy/medspacy#194] [General] [answered]
I was just wondering what dataset was medspacy trained on? There doesn't seem to be NER medical datasets that I can find.

**Comment by burgersmoke** [comment, marked as answer]
Good question @Samuel-Fipps.  I'll try to give some background here and give some thoughts about your discussion topic here.

As of this moment with medspacy 1.0.1, there are no trained language models which are distributed with medspacy.  Other systems might distribute models which are trained for NER or other clinical tasks, but the current incarnation of medspacy is a bit different.  While we might release some machine learned models in the future trained on synthetic or real clinical data, the current medspacy release contains _rules_ which were developed from various projects we have worked on as a part of Veterans Affairs or the University of Utah.  

These projects might have developed these _rules_ after developing using data from Veterans Affairs, University of Utah, the MIMIC-III dataset and in some case other clinical facilities.

So while there are some examples of concepts, sections, and other such things from clinical text, as of this moment, medspacy is meant to be more of a toolkit for developing clinical pipelines.  The idea is that developers might develop their own clinical concepts (TargetMatcher) or use concepts from UMLS (QuickUMLS), their own section rules, and contextual attributes of concepts (Context).

Stay tuned for future releases as we may release models trained on data, but think of this more as a toolkit for building high performance clinical pipelines, but not an end-to-end-solution.

Finally, as far as NER medical datasets, there are some out there, but indeed given privacy concerns and restrictions, they are less rare compared to other NER datasets in general domains like newswire text or social media  I will recommend a few medical (some biological, some clinical, some scientific research) related resources here.  Some of these include some NER datasets, but I will leave that up to you and others to dig deeper:

https://www.i2b2.org/NLP/DataSets/Main.php

https://n2c2.dbmi.hms.harvard.edu/data-sets

https://www.kaggle.com/datasets/finalepoch/medical-ner

https://figshare.com/articles/dataset/MACCROBAT2018/9764942
